File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

map<string,int>mp;
map<string,int>ans;
map<string,int>ch1;
map<string,int>ch2;

vector<int> solution(vector<string> id_list, vector<string> report, int k) {
    vector<int> answer;
    
    for(int i=0;i<report.size();i++){
        string s= "";
        for(int j=0;j<report[i].size();j++){
            if(report[i][j]==' ') {
                s="";
                continue;
            }
            s+=report[i][j];
        }
        if(ch1[report[i]]==0){
            ch1[report[i]]=1;
            mp[s]++;
        }
    }
    
    for(int i=0;i<report.size();i++){
        string s1= "";
        string s= "";
        for(int j=0;j<report[i].size();j++){
            if(report[i][j]==' ') {
                s1=s;
                s="";
                continue;
            }
            s+=report[i][j];
        }
         if(ch2[report[i]]==0){
            ch2[report[i]]=1;
            if(mp[s]>=k) ans[s1]++;
        }
    }
    
    for(string s : id_list){
        answer.push_back(ans[s]);
    }
    
    
    return answer;
}

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.java
import java.util.*;

class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        
        HashMap<String, Integer> hm = new HashMap<>();
        HashSet<String> hs = new HashSet<>();
        Set<String> list = new HashSet<>();
        
        for(int i=0;i<report.length;i++){
            String[] parts = report[i].split(" ");
            if(!hs.contains(parts[0]+" "+parts[1])){
                hs.add(parts[0]+" "+parts[1]);
                hm.put(parts[1], hm.getOrDefault(parts[1],0)+1);
            }
        }
        
        for(String s : hm.keySet()){
            if(hm.get(s) >=k) list.add(s);
        }
        
        List<Integer> answer = new ArrayList<>();
        
        for(int j=0;j<id_list.length;j++){
            int cnt=0;
            for(String s : list){
                if(hs.contains(id_list[j]+" "+s)) cnt++;
            }
            answer.add(cnt);
        }
        
        
        
        
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}